src/chunkserver/chunkserver_entry.h

```diff
@@ -0,0 +1,367 @@
+	/// another chunkserver for further processing.
+	void forward();
+
+	/// Initializes the connection to the next chunkserver in the chain.
+	///
+	/// This function sets up the necessary parameters and state for
+	/// establishing a connection to the next chunkserver.
+	///
+	/// This function is typically called when a new connection needs to be made
+	/// to forward data to another chunkserver.
+	///
+	/// @return An integer status code indicating the success or failure of the
+	///         connection initialization. A return value of 0 indicates
+	///         success, while a non-zero value indicates an error.
+	int initConnection();
+
+	/// Attempts to re-establish a connection to the next chunkserver.
+	/// Implements a retry mechanism to ensure that the connection
+	/// is eventually established
+	void retryConnect();
+
+	/// Checks and processes the next packet in the input buffer.
+	void checkNextPacket();
+
+	/// Processes a received packet based on its type.
+	///
+	/// @param type The type of the packet.
+	/// @param data Pointer to the packet data.
+	/// @param length The length of the packet data.
+	void gotPacket(uint32_t type, const uint8_t *data, uint32_t length);
+
+	/* IDLE Operations */
+
+	/// Answers to a ping message with the given data and length.
+	void ping(const uint8_t *data, PacketHeader::Length length);
+
+	/// Initializes a read operation
+	///
+	/// @param data Pointer to the buffer containing the information to read.
+	/// @param type The type of the packet.
+	/// @param length The length of the packet data.
+	void readInit(const uint8_t *data, PacketHeader::Type type,
+	              PacketHeader::Length length);
+
+	/// Continues a previously started read operation.
+	///
+	/// Processes the remaining data to be read from the chunkserver. If all
+	/// data has been read, it sends a read status message and closes the chunk.
+	/// Otherwise, it prepares the next part of the read operation.
+	///
+	/// @see ChunkserverEntry::readInit
+	void readContinue();
+
+	/// Requests a data prefetch operation.
+	/// Prefetch in this context means reading data from the disk and storing it
+	/// in the page cache.
+	void prefetch(const uint8_t *data, PacketHeader::Type type,
+	              PacketHeader::Length length);
+
+	/// Callback for when a read operation finishes.
+	static void readFinishedCallback(uint8_t status, void *entry);
+	/// Callback after delayed close operations.
+	static void delayedCloseCallback(uint8_t status, void *entry);
+	/// Callback for when a write operation finishes.
+	static void writeFinishedCallback(uint8_t status, void *entry);
+	/// Callback for legacy chunk block retrieval completion.
+	static void sauGetChunkBlocksFinishedLegacyCallback(uint8_t status,
+	                                                    void *entry);
+	/// Callback for chunk block retrieval completion.
+	static void sauGetChunkBlocksFinishedCallback(uint8_t status, void *entry);
```

Comment:
```suggestion
	static void getChunkBlocksFinishedCallback(uint8_t status, void *entry);
```
Consider removing the `sau` prefix to match other method naming convention inside the same object

Reply:
Not in the scope of this change.
will be handled as part of the refactoring, but this change is only moving to new files to facilitate the review.